lib/sycamore/sycamore/transforms/detr_partitioner.py

```diff
@@ -168,6 +244,276 @@ def partition_pdf(
         LogTime("finish", point=True)
         return deformable_layout
 
+    def _partition_pdf_batched(
+        self,
+        file: BinaryIO,
+        threshold: float = 0.4,
+        use_ocr=False,
+        ocr_images=False,
+        ocr_tables=False,
+        extract_table_structure=False,
+        table_structure_extractor=None,
+        extract_images=False,
+        batch_size: int = 1,
+    ) -> List[List["Element"]]:
+        LogTime("partition_start", point=True)
+        with tempfile.NamedTemporaryFile(prefix="detr-pdf-input-") as pdffile:
+            with LogTime("write_pdf"):
+                data = file.read()
```

Comment:
Is this ever an actual file, making this code redundant?

Reply:
it may be a file or may not. Eventually we'll fix the APIs to call the named version if you have a name.